hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/AbstractS3AMockTest.java

```diff
@@ -59,12 +59,14 @@ public void setup() throws Exception {
     Configuration conf = new Configuration();
     conf.setClass(S3_CLIENT_FACTORY_IMPL, MockS3ClientFactory.class,
         S3ClientFactory.class);
-    // We explicitly disable MetadataStore even if it's configured. For unit
-    // test we don't issue request to AWS DynamoDB service.
-    conf.setClass(S3_METADATA_STORE_IMPL, NullMetadataStore.class,
+    // We explicitly use local MetadataStore even if something else is
+    // configured. For unit test we don't issue request to AWS DynamoDB service
+    conf.setClass(S3_METADATA_STORE_IMPL, LocalMetadataStore.class,
```

Comment:
IMO this should only be set when -Ds3guard is passed on the command line, similar to what other tests were doing before. Tests that are only meaningful with S3Guard should Assume.assumeTrue() on this, but other tests that are meaningful without it as well should still be getting exercised.

Reply:
I'm good with that.  I'll fix this.

Reply:
I'm still planning to come back to this soon.

Reply:
Looking at this again, TestObjectChangeDetectionAttributes was the test that needed this change.  It's a unit test and the -Ds3guard and such stuff doesn't (at least currently) apply to unit tests.

I went a little different route where I switched AbstractS3AMockTest back to using NullMetdataStore by default but allowed override via overriding a createConfiguration() method like exists in some of the other base test classes.

TestObjectChangeDetectionAttributes now overrides createConfiguration() to use LocalMetadataStore.  While I was there, I noticed a deficiency that TestObjectChangeDetectionAttributes wouldn't really cover both change detection sources (ETag, versionId) so I updated it, making it parameterized and always covering both.